diff -uNr germinate.no-patch/germinate/archive.py germinate.patched/germinate/archive.py
--- germinate.no-patch/germinate/archive.py	2012-08-26 00:08:32.654670707 +0200
+++ germinate.patched/germinate/archive.py	2012-08-26 00:24:58.171557621 +0200
@@ -69,7 +69,7 @@
     """Fetch package lists from a Debian-format archive as apt tag files."""
 
     def __init__(self, dists, components, arch, mirrors, source_mirrors=None,
-                 installer_packages=True, cleanup=False):
+                 installer_packages=True, cleanup=False, archive_exceptions=[]):
         """Create a representation of a Debian-format apt archive."""
         if isinstance(dists, basestring):
             dists = [dists]
@@ -90,9 +90,10 @@
         else:
             self._source_mirrors = mirrors
         self._cleanup = cleanup
+        self._archive_exceptions = archive_exceptions
 
     def _open_tag_files(self, mirrors, dirname, tagfile_type,
-                        dist, component, ftppath):
+                        dist, component, ftppath, archive_exceptions=[]):
         def _open_tag_file(mirror, suffix):
             """Download an apt tag file if needed, then open it."""
             if not mirror.endswith('/'):
@@ -162,8 +163,20 @@
             return open(fullname, "r")
 
         tag_files = []
+	some_mirrors_processed = False
         for mirror in mirrors:
             tag_file = None
+            skip_this_repo = False
+            for archive in archive_exceptions:
+                if archive == mirror+","+dist+","+component or archive == mirror+","+dist+","+tagfile_type or archive == mirror+","+dist or archive == mirror:
+                    skip_this_repo = True
+                    break
+
+            if skip_this_repo:
+                print("Archive exception: skipping", mirror+","+dist+","+component)
+                continue
+            else:
+                some_mirrors_processed = True
             for suffix in (".bz2", ".gz", ""):
                 try:
                     tag_file = _open_tag_file(mirror, suffix)
@@ -171,7 +184,7 @@
                     break
                 except (IOError, OSError):
                     pass
-        if len(tag_files) == 0:
+        if some_mirrors_processed and len(tag_files) == 0:
             raise IOError("no %s files found" % tagfile_type)
         return tag_files
 
@@ -194,30 +207,35 @@
             for component in self._components:
                 packages = self._open_tag_files(
                     self._mirrors, dirname, "Packages", dist, component,
-                    "binary-" + self._arch + "/Packages")
+                    "binary-" + self._arch + "/Packages", self._archive_exceptions)
                 for tag_file in packages:
                     try:
                         for section in apt_pkg.TagFile(tag_file):
                             yield (IndexType.PACKAGES, section)
                     finally:
                         tag_file.close()
-
-                sources = self._open_tag_files(
-                    self._source_mirrors, dirname, "Sources", dist, component,
-                    "source/Sources")
-                for tag_file in sources:
-                    try:
-                        for section in apt_pkg.TagFile(tag_file):
-                            yield (IndexType.SOURCES, section)
-                    finally:
-                        tag_file.close()
+                try:
+                    sources = self._open_tag_files(
+                        self._source_mirrors, dirname, "Sources", dist, component,
+                        "source/Sources", self._archive_exceptions)
+                except IOError:
+                        # can live without these
+                        _progress("Missing Source Packages file for %s "
+                                  "(ignoring)", component)
+                else:
+                    for tag_file in sources:
+                        try:
+                            for section in apt_pkg.TagFile(tag_file):
+                                yield (IndexType.SOURCES, section)
+                        finally:
+                            tag_file.close()
 
                 instpackages = ""
                 if self._installer_packages:
                     try:
                         instpackages = self._open_tag_files(
                             self._mirrors, dirname, "InstallerPackages", dist, component,
-                            "debian-installer/binary-" + self._arch + "/Packages")
+                            "debian-installer/binary-" + self._arch + "/Packages", self._archive_exceptions)
                     except IOError:
                         # can live without these
                         _progress("Missing installer Packages file for %s "
diff -uNr germinate.no-patch/germinate/scripts/germinate_update_metapackage.py germinate.patched/germinate/scripts/germinate_update_metapackage.py
--- germinate.no-patch/germinate/scripts/germinate_update_metapackage.py	2012-08-26 00:08:32.658670716 +0200
+++ germinate.patched/germinate/scripts/germinate_update_metapackage.py	2012-08-26 00:20:38.954272229 +0200
@@ -128,6 +128,15 @@
     except (NoSectionError, NoOptionError):
         archive_base_default = None
 
+    try:
+        archive_dists = config.get(dist, 'dists').split()
+    except NoOptionError:
+        archive_dists = [dist]
+    try:
+       archive_exceptions = config.get(dist, 'archive_base/exceptions').split()
+    except (NoSectionError, NoOptionError):
+       archive_exceptions = []
+
     archive_base = {}
     for arch in architectures:
         try:
@@ -263,9 +272,9 @@
         print("[%s] Downloading available package lists..." % architecture)
         germinator = Germinator(architecture)
         archive = germinate.archive.TagFile(
-            [dist], components, architecture,
+            archive_dists, components, architecture,
             archive_base[architecture], source_mirrors=archive_base_default,
-            cleanup=True)
+            cleanup=True, archive_exceptions=archive_exceptions)
         germinator.parse_archive(archive)
         debootstrap_base = set(debootstrap_packages(architecture))
 
